testsuite/MDAnalysisTests/utils/test_imports.py

```diff
@@ -22,36 +22,42 @@
 from __future__ import absolute_import
 
 import os
+
 import MDAnalysisTests
+import pytest
+
+"""Relative imports are banned in unit testing modules (Issue #189), so run tests to enforce this policy."""
+
+path_to_testing_modules = MDAnalysisTests.__path__[0]
+# Exclusion path relative to MDAnalysisTests
+exclusions = ['/plugins']
+
 
-class TestRelativeImports(object):
-    """Relative imports are banned in unit testing modules (Issue #189), so run tests to enforce this policy."""
-
-    path_to_testing_modules = MDAnalysisTests.__path__[0]
-    # Exclusion path relative to MDAnalysisTests
-    exclusions = ['/plugins']
-
-    @classmethod
-    def is_excluded(cls, path):
-        leaf = path[len(cls.path_to_testing_modules):]
-        return leaf in cls.exclusions
-
-    @staticmethod
-    def _run_test_relative_import(testing_module):
-        with open(testing_module, 'r') as test_module_file_object:
-            for lineno, line in enumerate(test_module_file_object, start=1):
-                if 'from .' in line and 'import' in line \
-                        and not 'test_imports' in testing_module:
-                    raise AssertionError(
-                        "A relative import statement was found in "
-                        "module {testing_module} at linenumber {lineno}.".format(**vars()))
-
-    def test_relative_imports(self):
-        for dirpath, dirnames, files in os.walk(self.path_to_testing_modules):
-            if self.is_excluded(dirpath):
+def is_excluded(path):
+    leaf = path[len(path_to_testing_modules):]
+    return leaf in exclusions
+
+
+def get_file_paths():
+    paths = []
+    for dirpath, dirnames, files in os.walk(path_to_testing_modules):
+        if is_excluded(dirpath):
+            continue
+        for f in filter(lambda x: x.endswith('.py'), files):
+            fpath = os.path.join(dirpath, f)
+            if is_excluded(fpath):
                 continue
-            for f in filter(lambda x: x.endswith('.py'), files):
-                fpath = os.path.join(dirpath, f)
-                if self.is_excluded(fpath):
-                    continue
-                yield self._run_test_relative_import, fpath
+            # yield self._run_test_relative_import, fpath
+            paths.append(fpath)
+    return paths
+
+
+@pytest.mark.parametrize('testing_module', get_file_paths())
```

Comment:
looks like a sensible and pretty clean way to turn the yield-based test into a parametrized one, good!